### PR TITLE
Quin Nordmark Pull Request

### PR DESCRIPTION
**Pull request by n0rdie**

### CODE REVIEW

File: lib/budget.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+require './lib/employee'
+require './lib/department'
```
Comment (KatBrandt) [inline lib/budget.rb:2]: These requires should only be in the spec files.

File: lib/employee.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+    def initialize(input_info)
+        @name = input_info[:name]
+        @age = input_info[:age].to_i
+        @salary = input_info[:salary][1..-1].to_i
+    end
+
+    def give_raise(raise_ammount)
+        @salary += raise_ammount
+    end
```
Comment (KatBrandt) [inline lib/employee.rb:12]: Great job with this class. All looks great.

File: spec/employee_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        expect(bobbi.age).to eq(30)
+        expect(bobbi.salary).to eq(100000)
+    end
+
+    it "can give raises" do
+        bobbi = Employee.new({name: "Bobbi Jaeger", age: "30", salary: "$100000"})
+        bobbi.give_raise(5000)
+        
+        expect(bobbi.salary).to eq(105000)
+    end
```
Comment (KatBrandt) [inline spec/employee_spec.rb:18]: Yup, good testing here with this file.

File: lib/department.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+require './lib/employee'
+
+class Department
+    attr_reader :employees, :expenses, :name
+
+    def initialize(name_input)
```
Comment (KatBrandt) [inline lib/department.rb:6]: You could just call this argument `name`

File: spec/department_spec.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+    it "can add new expenses" do
+        customer_service = Department.new("Customer Service")
+        expect(customer_service.expenses).to eq(0)
+
+        customer_service.expense(100)
+        expect(customer_service.expenses).to eq(100)
+
+        customer_service.expense(25)
+        expect(customer_service.expenses).to eq(125)
```
Comment (KatBrandt) [inline spec/department_spec.rb:31]: Great test here showing we can add one expense, then another, and we can see that our method is tracking that correctly.

File: lib/budget.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    end
+
+    def departments_with_low_expenses
+        cheap_departments = []
+            @departments.each do |department|
+                if department.expenses < 500
+                    cheap_departments.append(department)
+                end
+            end
+        cheap_departments
```
Comment (KatBrandt) [inline lib/budget.rb:24]: Great use of .each here! What other enumerable could you use to condense this code?

File: lib/budget.rb [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    end
+
+    def employee_salaries
+        salaries = []
+            @departments.each do |department|
+                department.employees.each do |employee|
+                    salaries.append(employee.salary)
+                end
+            end
+        salaries
```
Comment (KatBrandt) [inline lib/budget.rb:34]: Same as above, great use of .each. I'd love to see you expand your enumerable usage.

File: lib/budget.rb [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    end
+
+    def current_expenses_by_department
+        hash = {}
+        @departments.each do |department|
+            hash.merge!({department=>department.expenses})
```
Comment (KatBrandt) [inline lib/budget.rb:40]: Generally speaking, I like to stay away from methods that use the bang operator as they can be destructive. This definitely works. Another way to do this would be to have line 40 be
`hash[department] = department.expenses`
Also, I'd love to see the hash named not `hash`, but something more indicative of the data it's holding or accumulating.

File: spec/budget_spec.rb [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+        department1 = Department.new("Towing Crew")
+        department1.expense(600)
+        budget.add_department(department1)
+
+        department2 = Department.new("Reposession Team")
+        department2.expense(300)
+        budget.add_department(department2)
+        
+        expect(budget.departments_with_low_expenses).to eq([department2])
```
Comment (KatBrandt) [inline spec/budget_spec.rb:33]: Great test. To make this more robust, you can then add another low-expenses department and make sure your array now holds 2 departments. That way we ensure it finds ALL the low-expenses departments.

File: spec/budget_spec.rb [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+        department1.hire(tony)
+        department1.hire(stevie)
+        budget.add_department(department1)
+
+        department2 = Department.new("Reposession Team")
+        department1.hire(marcello)
+        department1.hire(luigi)
+        budget.add_department(department2)
+        
+        expect(budget.employee_salaries).to eq([79000, 68000, 120000, 82000])
```
Comment (KatBrandt) [inline spec/budget_spec.rb:54]: Great!

File: spec/budget_spec.rb [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+        department1 = Department.new("Towing Crew")
+        department1.expense(600)
+        budget.add_department(department1)
+
+        department2 = Department.new("Reposession Team")
+        department2.expense(300)
+        budget.add_department(department2)
+        
+        expect(budget.current_expenses_by_department[department1]).to eq(600)
+        expect(budget.current_expenses_by_department[department2]).to eq(300)
```
Comment (KatBrandt) [inline spec/budget_spec.rb:71]: This is a good test, but it doesn't ensure that this is the ONLY data in the hash. You can actually build the hash out literally in the test and then test that your return value matches that hash exactly.

**Review by KatBrandt — COMMENTED** [review]
Overall, really great work here Quin! Your testing is looking good and your implementation code is looking great. I encourage you to stretch yourself by thinking about more robust testing, edge case testing, and using more advanced enumerables.
